packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -100,17 +106,107 @@ export interface ISummarizerEvents extends IEvent {
      */
     (event: "summarizingError", listener: (error: ISummarizingWarning) => void);
 }
+export type SummarizerStopReason =
+    /** Summarizer client failed to summarize in all 3 consecutive attempts. */
+    | "failToSummarize"
+    /**
+     * Summarizer client detected that its parent is no longer elected the summarizer.
+     * Normally, the parent client would realize it is disconnected first and call stop
+     * giving a "parentNotConnected" stop reason. If the summarizer client attempts to
+     * generate a summary and realizes at that moment that the parent is not elected,
+     * only then will it stop itself with this message.
+     */
+    | "parentNoLongerSummarizer"
+    /** Parent client reported that it is no longer connected. */
+    | "parentNotConnected"
+    /**
+     * Parent client reported that it is no longer elected the summarizer.
+     * This is the normal flow; a disconnect will always trigger the parent
+     * client to no longer be elected as responsible for summaries. Then it
+     * tries to stop its spawned summarizer client.
+     */
+    | "parentShouldNotSummarize"
+    /** Parent client reported that it is disposed. */
+    | "disposed";
 export interface ISummarizer
     extends IEventProvider<ISummarizerEvents>, IFluidRouter, IFluidRunnable, IFluidLoadable {
     /**
      * Returns a promise that will be resolved with the next Summarizer after context reload
      */
     setSummarizer(): Promise<Summarizer>;
-    stop(reason?: string): void;
+    stop(reason?: SummarizerStopReason): void;
     run(onBehalfOf: string): Promise<void>;
     updateOnBehalfOf(onBehalfOf: string): void;
 }
 
+type SummarizeReason =
+    /**
+     * Attempt to summarize after idle timeout has elapsed.
+     * Idle timer restarts whenever an op is received. So this
+     * triggers only after some amount of time has passed with
+     * no ops being received.
+     */
+    | "idle"
+    /**
+     * Attempt to summarize after a maximum time since last
+     * successful summary has passed. This measures time since
+     * last summary ack op was processed.
+     */
+    | "maxTime"
+    /**
+     * Attempt to summarize after a maximum number of ops have
+     * passed since the last successful summary. This compares
+     * op sequence numbers with the reference sequence number
+     * of the summarize op corresponding to the last summary
+     * ack op.
+     */
+    | "maxOps"
+    /**
+     * Special case to generate a summary immediately after starting.
+     * @deprecated - only used in context reload hot swap, which was removed
+     */
+    | "immediate"
```

Comment:
Why not remove deprecated items from this list? It does not seem like there are any references to it in the code.
Similar for save op, though if we leave it, can you add prefix to the message indicating it's because of save op?

Reply:
Added "save;" prefix. And removed "immediate". I think a separate PR could probably just fully remove save ops from the system.